src/content-script/listeners/player.ts

```diff
@@ -0,0 +1,110 @@
+import { debug } from '@root/lib/utils';
+import socket from '@root/lib/socket';
+import { SocketEvents } from '@root/lib/constants/socket';
+import store from '@contentScript/store';
+
+//------------------------------------------------------------------------------
+
+// functions for each platform that can be run to get their video
+const videoLocators = {
+  youtube: () => {
+    return document.querySelector('.html5-main-video');
+  },
+  vimeo: () => {
+    return document.querySelector('video');
+  },
+  default: () => {
+    return document.querySelector('video');
+  },
+};
+
+//------------------------------------------------------------------------------
+
+function findVideoPlayer() {
+  const url = window.location.href;
+  const platforms = ['youtube', 'vimeo'];
+
+  let player = null;
+
+  for (let i = 0; i < platforms.length; i += 1) {
+    const platform = platforms[i];
+    if (url.includes(platform)) {
+      debug(`Located ${platform} video element`);
+      player = videoLocators[platform]();
+      debug(player);
+    }
+  }
+
+  // Unknown site, try to get video anyway
+  if (!player) player = videoLocators.default();
+
+  // Still no player? Fail
+  if (!player) {
+    debug("Couldn't find video player");
+    // TODO: register some state for the UI to display the issue
+  }
+
+  return player;
+}
+
+//------------------------------------------------------------------------------
+
+function extractEventData(event) {
+  // Need to evaluate non-enumerables: https://stackoverflow.com/a/48055076
+  function evalSingleProp(parent, prop) {
+    Object.defineProperty(parent, prop, {
+      value: parent[prop],
+      enumerable: true,
+      configurable: true,
+    });
+  }
+
+  evalSingleProp(event, 'srcElement');
+
+  const props = ['currentTime', 'duration', 'playbackRate'];
+  props.forEach(prop => {
+    evalSingleProp(event.srcElement, prop);
+  });
+
+  return event.target;
```

Comment:
I get what you're trying to do here but we can shorten this down to one line of code 😁. Looks like those props aren't enumerable thus unserializable. However, we can just copy them over to a new object.
```suggestion
const { currentTime, duration, playbackRate } = event.target;
return { currentTime, duration, playbackRate };
```

We should install `lodash`, and reduce this more:

Can you install? Run the following:

```
yarn add lodash @types/lodash
```

Then import pick from lodash, and the function simply becomes:

```ts
return pick(event.target, ['currentTime', 'duration', 'playbackRate']);
```

Reply:
Are you trying to say I shouldn't just blindly do whatever stack-overflow tells me!? :)
Good change, skipping lodash for now though